korman/properties/modifiers/render.py

```diff
@@ -500,3 +500,35 @@ def export(self, exporter, bo, so):
             if not region.control_region:
                 raise ExportError("{}: Not all Visibility Controls are set up properly in Visibility Set".format(bo.name))
             addRegion(exporter.mgr.find_create_key(plVisRegion, bl=region.control_region))
+
+
+fog_types = {
+    "none":     plFogEnvironment.kNoFog,
+    "linear":   plFogEnvironment.kLinearFog,
+    "exp2":     plFogEnvironment.kExp2Fog
+}
+
+class PlasmaFogEnvMod(PlasmaModifierProperties):
+    pl_id = "fogenv"
+
+    bl_category = "Render"
+    bl_label = "Fog Environment"
+    bl_description = "Adjust per-object fog settings"
+    bl_icon = "MAT_SPHERE_SKY"
+
+    fog_env = PointerProperty(name="Environment",
+                              description="Environment defining the fog settings for this object",
+                              type=bpy.types.World)
+
+    def export(self, exporter, bo, so):
+        if self.fog_env is None or self.fog_env == bpy.context.scene.world:
+            return # Don't generate a FogEnv for the default Age FNI fog
+
+        fe = exporter.mgr.find_create_object(plFogEnvironment, bl=bo, name=self.fog_env.name)
+        env = self.fog_env.plasma_fni
+
+        fe.type = fog_types[env.fog_method]
```

Comment:
The preferred method for doing this is to have blender store the attribute name of the constant and use `getattr` to apply it.

Reply:
I don't think I can do that in this case because the `plasma_fni` enum values don't match the names of the constants, but are already in use in existing data :(